code/processes/metrics.q

```diff
@@ -35,49 +41,39 @@ metrics:{[syms]
    / add allday window
    if[.metrics.enableallday;
     if[not all syms in key .metrics.start;.metrics.start::exec first time by sym from sumstab];
-    t:`sym`timediff xasc t,select sym,timediff:0Nn,vwap:sumsps%sumssize,twap:sumspricetimediff%.z.p - .metrics.start[sym] from latest where sym in syms];
-
-   t
-  
+    t:`sym`timediff xasc t,select sym,timediff:0Nn,vwap:sumsps%sumssize,twap:sumspricetimediff%.z.p - .metrics.start[sym] from latest where sym in syms];t  
  }
 
 \d .metrics 
 
 / check for TP connection
 notpconnected:{[]
-	0 = count select from .sub.SUBSCRIPTIONS where proctype in ((),`tickerplant), active}
+	0 = count select from .sub.SUBSCRIPTIONS where proctype in tickerplanttypes,active}
 
 / get handle for TP & subscribe
 subscribe:{
-  / get handle
-  if[count s:.sub.getsubscriptionhandles[`tickerplant;();()!()];
-  subproc:first s;
-  / if got handle successfully, subsribe to trade table
-  .lg.o[`subscribe;"subscribing to ", string subproc`procname];
-  :.sub.subscribe[`trade;`;0b;0b;subproc]]};
+   / get handle
+   if[count s:.sub.getsubscriptionhandles[`tickerplant;();()!()];
+     subproc:first s;
+     / if got handle successfully, subsribe to trade table
+     .lg.o[`subscribe;"subscribing to ", string subproc`procname];
+     :.sub.subscribe[`trade;`;0b;0b;subproc]]
+ };
 
 / get subscribed to TP, recover up until now from RDB
 init:{
   r:subscribe[];
 
-  / make sure connection to TP was successful, or else wait
-  while[notpconnected[];
-   / wait 10 seconds
-   .os.sleep[10];
-   / try again to connect to discovery
-   .servers.startup[];
-   / try again to subscribe to TP
-   r:subscribe[]];
+  // Block process until all required processes are connected
+  r:.servers.startupdependent[requiredprocs;tpconsleep;cycles;string subscribe;`metrics]; 
 
   / check if updates have already been sent from TP, if so recover from RDB
-  if[r[`icounts][`trade] > 0;
-   / make sure connection is made to RDB
-   while[not count s:.sub.getsubscriptionhandles[`rdb;();()!()];.os.sleep[10]];
+  if[r[`icounts]`trade >0;
```

Comment:
Does this definitely work out the same? May need to be:
```
if[0<r[`icounts]`trade;
```

Reply:
indeed. I changed it now.